src/services/sync/strategies/LDAPSchoolSyncer.js

```diff
@@ -52,10 +52,39 @@ class LDAPSchoolSyncer extends Syncer {
 		this.logInfo('Getting users...');
 		const ldapUsers = await this.app.service('ldap').getUsers(this.system.ldapConfig, this.school);
 		this.logInfo(`Creating and updating ${ldapUsers.length} users...`);
-		for (const ldapUser of ldapUsers) {
+
+		const bulkSize = 1000; // 5000 is a hard limit because of definition in user model
+
+		for (let i = 0; i < ldapUsers.length; i += bulkSize) {
+			const ldapUserChunk = ldapUsers.slice(i, i + bulkSize);
+			const ldapUserIds = ldapUserChunk.map((ldapUser) => ldapUser.ldapUUID);
 			// eslint-disable-next-line no-await-in-loop
-			await this.createOrUpdateUser(ldapUser);
-			this.updateModifyTimestampMaximum(ldapUser.modifyTimestamp);
+			const userData = await this.app.service('usersModel').find({
+				query: {
+					ldapId: { $in: ldapUserIds },
+					$populate: ['roles'],
+					$limit: bulkSize, // needed for bulkSize > 1000 because of default limit
+				},
+			});
+
+			for (const ldapUser of ldapUserChunk) {
+				try {
+					const dbUser = userData.data.find((userToFind) => userToFind.ldapId === ldapUser.ldapUUID);
```

Comment:
Just an idea, could we improve the find performance by sorting the data array by ldapId before entering the loop?

Reply:
I thought about it. This is O(m*n). Sorting first would be O(m*log(m)+n).
The difference is very small. But it is harder to understand.
And the computational time needed here compared to the time of the db requests is probably 1:100.